### PR TITLE
Converted stateful class components to function components using hooks

### PR DESCRIPTION
**Pull request by Mariela-t**

### CODE REVIEW

File: src/NumberGuessingGame.js [hunk 1]
```diff
@@ -13,65 +13,43 @@ function getRandomNumber() {
 
 const MAX_ATTEMPTS = 5;
 
-class NumberGuessingGame extends Component {
-  constructor(props) {
-    super(props);
+const NumberGuessingGame = () => {
+  const [numberToGuess, setNumberToGuess] = useState(getRandomNumber());
```
Comment (kwilson05) [inline src/NumberGuessingGame.js]: This works for the most part but `numberToGuess` is actually initialized twice.

Try to figure out a way to initialize it once 👍

File: src/NumberGuessingGame.js [hunk 2]
```diff
@@ -13,65 +13,43 @@ function getRandomNumber() {
...
+  const isCorrectGuess = latestGuess === numberToGuess;
 
-    /**
-     * These lines are required to make the methods/functions declared on this
-     *  class have the correct `this` object when they run.
-     */
-    this.handleGuess = this.handleGuess.bind(this);
-    this.handleReset = this.handleReset.bind(this);
-  }
+  const isGameOver = isCorrectGuess || numberOfGuesses === MAX_ATTEMPTS;
```
Comment (kwilson05) [inline src/NumberGuessingGame.js:22]: 💯 I like this

**Review by kwilson05 — APPROVED** [review]
Good job!! 💯 
Everything works here but try an address the one slight but then merge away 🎖️

File: src/NumberGuessingGame.js (hunk 1 above)
Comment (Mariela-t) [reply src/NumberGuessingGame.js:17]: I looked over my code but I only saw numberToGuess initialized once through the useState hook. I am not sure if I overlooked something?

**Pull request merged**